contingency/contingency-api/src/main/java/com/powsybl/contingency/CriterionContingencyList.java

```diff
@@ -0,0 +1,70 @@
+/**
+ * Copyright (c) 2022, RTE (http://www.rte-france.com)
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+package com.powsybl.contingency;
+
+import com.google.common.collect.ImmutableList;
+import com.powsybl.contingency.contingency.list.criterion.Criterion;
+import com.powsybl.iidm.network.*;
+
+import java.util.List;
+import java.util.Objects;
+import java.util.stream.Collectors;
+
+import static com.powsybl.contingency.Contingency.getContingencyElement;
+
+/**
+ * @author Etienne Lesot <etienne.lesot@rte-france.com>
+ */
+public class CriterionContingencyList implements ContingencyList {
+
+    // VERSION = 1.0 : first version
+    public static final String VERSION = "1.0";
+
+    private final String name;
+    private final IdentifiableType identifiableType;
+    private final List<Criterion> criteria;
```

Comment:
After some thinking, I tend to think it will be better to hard code in the fields the expected criterions (instead of having an arbitrary list), because for example we will never want 2 criterions on countries, or 2 criterions on voltage.

@geofjamg : any opinion about this design ?

Note that it does not prevent to use a List<Criterion> to actually perform the filtering.

Then if we go for this, the next question was "do we want to have different implementations for different types", to be able to have well defined fields for each type. Same thing, I tend to think that yes, now ... But it means one implementation for each type of element, not any more one for all.

In that case, we could keep this one as an abstract class.